test/Riok.Mapperly.Tests/Mapping/ObjectPropertyTest.cs

```diff
@@ -303,4 +303,56 @@ public Task WithNotFoundIgnoredSourcePropertyShouldDiagnostic()
 
         return TestHelper.VerifyGenerator(source);
     }
+
+    [Fact]
+    public Task WithPrivateTargetSetterShouldIgnoreAndDiagnostic()
+    {
+        var source = TestSourceBuilder.Mapping(
+            "A",
+            "B",
+            "class A { public string StringValue { get; set; } public int IntValue { get; private set; } }",
+            "class B { public string StringValue { get; private set; } public int IntValue { private get; set; } }");
+
+        return TestHelper.VerifyGenerator(source);
+    }
+
+    [Fact]
+    public Task WithPrivateTargetPathGetterShouldIgnoreAndDiagnostic()
+    {
+        var source = TestSourceBuilder.Mapping(
+            "A",
+            "B",
+            "class A { public C StringValue { private get; set; } public int IntValue { get; private set; } }",
+            "class B { public C StringValue { get; private set; } public int IntValue { private get; set; } }",
+            "class C { public string StringValue { get; set; } }",
+            "class D { public string StringValue { get; set; } }");
+
+        return TestHelper.VerifyGenerator(source);
+    }
+
+    [Fact]
+    public Task WithPrivateSourceGetterShouldIgnoreAndDiagnostic()
+    {
+        var source = TestSourceBuilder.Mapping(
+            "A",
+            "B",
+            "class A { public string StringValue { private get; set; } public int IntValue { get; private set; } }",
+            "class B { public string StringValue { get; set; } public int IntValue { private get; set; } }");
+
+        return TestHelper.VerifyGenerator(source);
+    }
+
+    [Fact]
+    public Task WithPrivateSourcePathGetterShouldIgnoreAndDiagnostic()
+    {
+        var source = TestSourceBuilder.Mapping(
+            "A",
+            "B",
+            "class A { public C StringValue { private get; set; } public int IntValue { get; private set; } }",
+            "class B { public C StringValue { get; set; } public int IntValue { private get; set; } }",
+            "class C { public string StringValue { get; set; } }",
+            "class D { public string StringValue { get; set; } }");
```

Comment:
Class `D` is unused here. In all the new test cases, I would use a name other than `StringValue` for the properties of type `C` and `D`. For example `Nested` or something else

Reply:
Improved.

Reply:
Now class `C` is unused in all tests

Reply:
🤦 should be correct now.